### PR TITLE
Common compatible suppression3

### PR DESCRIPTION
**Pull request by jtlap**
remains if_else, bit compatibility and the proba module

### CODE REVIEW

File: include/eve/module/combinatorial/regular/impl/fibonacci.hpp [hunk 1]
```diff
@@ -9,13 +9,14 @@
 
 #include <eve/module/core.hpp>
 #include <eve/module/math.hpp>
+#include <eve/traits/common_value.hpp>
```
Comment (jfalcou) [inline include/eve/module/combinatorial/regular/impl/fibonacci.hpp:12]: remove if not used

File: include/eve/module/combinatorial/regular/impl/fibonacci.hpp [hunk 2]
```diff
@@ -9,13 +9,14 @@
...
 
 namespace eve::detail
 {
 
 template<unsigned_value N, floating_value T, value U>
```
Comment (jfalcou) [inline include/eve/module/combinatorial/regular/impl/fibonacci.hpp:17]: check if this is coherent in another PR

File: include/eve/module/core/regular/impl/logical_andnot.hpp [hunk 3]
```diff
@@ -17,6 +16,7 @@
 #include <eve/module/core/regular/is_nez.hpp>
 #include <eve/traits/as_logical.hpp>
 #include <eve/traits/is_logical.hpp>
+#include <eve/traits/common_value.hpp>
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/logical_andnot.hpp:19]: check if necesary

File: include/eve/module/core/regular/impl/negminabs.hpp [hunk 4]
```diff
@@ -17,14 +16,16 @@ namespace eve::detail
 
 template<typename  ...Ts>
 EVE_FORCEINLINE auto
-negminabs_(EVE_SUPPORTS(cpu_), Ts... args)
+negminabs_(EVE_SUPPORTS(cpu_), Ts... args) noexcept
+-> decltype(minabs(args...))
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/negminabs.hpp:20]: change to `minus(minabs(...))` for coherence

File: include/eve/module/core/regular/impl/negminabs.hpp [hunk 5]
```diff
@@ -17,14 +16,16 @@ namespace eve::detail
...
 {
   return minus(minabs(args...));
 }
 
 template<conditional_expr C, typename  ...Ts>
 EVE_FORCEINLINE auto
-negminabs_(EVE_SUPPORTS(cpu_), C const & c, Ts... args)
+negminabs_(EVE_SUPPORTS(cpu_), C const & c, Ts... args) noexcept
+-> decltype(minabs(args...))
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/negminabs.hpp:28]: use `if_else`

File: include/eve/module/polynomial/detail/horner_impl.hpp [hunk 6]
```diff
@@ -24,6 +25,7 @@ namespace eve::detail
 template<decorator D, value T0, value... Cs>
 EVE_FORCEINLINE constexpr auto
 horner_impl(D const& d, T0 const& xx, Cs... cs) noexcept
+-> common_value_t<T0, Cs...>
 {
   using r_t          = common_compatible_t<T0, Cs...>;
```
Comment (jfalcou) [inline include/eve/module/polynomial/detail/horner_impl.hpp:30]: `using r_t          = common_compatible_t<T0, Cs...>;` 
should be
`using r_t          = common_value_t<T0, Cs...>;`

and then just use `auto`

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/polynomial/detail/horner_impl.hpp (hunk 6 above)
Comment (jtlap) [reply include/eve/module/polynomial/detail/horner_impl.hpp:30]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**